prow/flagutil/github.go

```diff
@@ -56,6 +60,27 @@ func (o *GitHubOptions) addFlags(wantDefaultGithubTokenPath bool, fs *flag.FlagS
 	}
 	fs.StringVar(&o.TokenPath, "github-token-path", defaultGithubTokenPath, "Path to the file containing the GitHub OAuth secret.")
 	fs.StringVar(&o.deprecatedTokenFile, "github-token-file", "", "DEPRECATED: use -github-token-path instead.  -github-token-file may be removed anytime after 2019-01-01.")
+	baseURL, err := getBaseURL(o.endpoint)
+
+	if err != nil {
+		logrus.Fatalf("could not get baseURL from endpoints: %v", err)
+	}
+	o.baseURL = baseURL
+}
+
+// getBaseURL infer base url from the github endpoint
+func getBaseURL(endpoints Strings) (string, error) {
+	for _, endpoint := range endpoints.vals {
+		endpointURL, err := url.ParseRequestURI(endpoint)
+		if err != nil {
+			return "", fmt.Errorf("invalid url format for: %q", endpoint)
+		}
+		if endpointURL.Host != "api.github.com" && endpointURL.Host != "ghproxy" {
```

Comment:
This is why I re-implemented, these conditionals could cause hard to debug issues with some of the following use cases:
- using `ghproxy` with github enterprise will not work
- using a different vhost for github enterprise api and storage
- using `ghproxy` at a different fqdn, ie: `proxy.ghproxy`, `gitproxy`, `ghproxy.mycorp.com`, `nginx`

This inference tactic also doesn't help us move to other SCM providers.